cocotb/drivers/avalon.py

```diff
@@ -512,9 +513,8 @@ class AvalonST(ValidatedBusDriver):
 
     _default_config = {"firstSymbolInHighOrderBits" : True}
 
-    def __init__(self, *args, **kwargs):
-        config = kwargs.pop('config', {})
-        ValidatedBusDriver.__init__(self, *args, **kwargs)
+    def __init__(self, entity, name, clock, *, config={}, **kwargs):
```

Comment:
This is python3-only syntax, and we haven't yet decided what to do for #767. At any rate, we should probably aim to do at least one more python 2 release.

Reply:
So the question is then what is preferred:
```python
    def __init__(self, entity, name, clock, config={}, **kwargs):
```
or
```python
    def __init__(self, entity, name, clock, **kwargs):
        config = kwargs.pop('config', {})
```
I like the former because then the parameter is shown when doing help in ipython etc. but then people could wrongly use positional passing of config.

Reply:
I would go for the latter - we can fix the docs as soon as we drop python 2, but we can't change the API once we release it.

Reply:
Will keep use of pop for the moment; can be reconsidered later.